platform/android/export/gradle_export_util.h

```diff
@@ -0,0 +1,101 @@
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef GODOT_GRADLE_EXPORT_UTIL_H
+#define GODOT_GRADLE_EXPORT_UTIL_H
+
+#include "core/io/zip_io.h"
+#include "core/os/dir_access.h"
+#include "core/os/file_access.h"
+#include "core/os/os.h"
+#include "editor/editor_export.h"
+
+// Static utility method used to create a directory.
+static Error create_directory(String dir) {
+	if (!DirAccess::exists(dir)) {
+		DirAccess *filesystem_da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+		ERR_FAIL_COND_V_MSG(!filesystem_da, ERR_CANT_CREATE, "Cannot create directory '" + dir + "'.");
+		Error err = filesystem_da->make_dir_recursive(dir);
+		ERR_FAIL_COND_V_MSG(err, ERR_CANT_CREATE, "Cannot create directory '" + dir + "'.");
+		memdelete(filesystem_da);
+	}
+	return OK;
+}
+
+// Implementation of EditorExportSaveSharedObject.
+// This method will only be called as an input to export_project_files.
+// This method lets the .so files for all ABIs to be copied
+// into the gradle project from the .AAR file
+static Error ignore_so_file(void *p_userdata, const SharedObject &p_so) {
+	return OK;
+}
+
+// Writes p_data into a file at p_path, creating directories if necessary.
+// Note: this will overwrite the file at p_path if it already exists.
+static Error store_file_at_path(const String &p_path, const Vector<uint8_t> &p_data, int compression_method = Z_DEFLATED) {
+	String dir = p_path.get_base_dir();
+	Error err = create_directory(dir);
+	if (err != OK) {
+		return err;
+	}
+	FileAccess *fa = FileAccess::open(p_path, FileAccess::WRITE);
+	ERR_FAIL_COND_V_MSG(!fa, ERR_CANT_CREATE, "Cannot create file '" + p_path + "'.");
+	fa->store_buffer(p_data.ptr(), p_data.size());
+	memdelete(fa);
+	return OK;
+}
+
+// Writes string p_data into a file at p_path, creating directories if necessary.
+// Note: this will overwrite the file at p_path if it already exists.
+static Error store_string_at_path(const String &p_path, const String &p_data, int compression_method = Z_DEFLATED) {
+	String dir = p_path.get_base_dir();
+	Error err = create_directory(dir);
+	if (err != OK) {
+		return err;
+	}
+	FileAccess *fa = FileAccess::open(p_path, FileAccess::WRITE);
+	ERR_FAIL_COND_V_MSG(!fa, ERR_CANT_CREATE, "Cannot create file '" + p_path + "'.");
+	fa->store_string(p_data);
+	memdelete(fa);
+	return OK;
+}
+
+// Implementation of EditorExportSaveFunction.
+// This method will only be called as an input to export_project_files.
+// It is used by the export_project_files method to save all the asset files into the gradle project.
+// It's functionality mirrors that of the method save_apk_file.
+// This method will be called ONLY when custom build is enabled.
+static Error rename_and_store_file_in_gradle_project(void *p_userdata, const String &p_path, const Vector<uint8_t> &p_data, int p_file, int p_total) {
```

Comment:
I'm not a C/C++ guru so this might be a silly question, but what's the impact of making this `static` in a header instead of having it non `static`?

Reply:
I modeled these methods off of methods like [`save_apk_so`](https://github.com/amanj120/godot/blob/3.2/platform/android/export/export.cpp#L825), which are static methods. Originally, the methods I laid out here were supposed to be in `platform/android/export/export.cpp`, but now that they're in a separate `.h` file, I see no reason to keep them static.

Reply:
@m4gr3d do you have any strong preference one way or another?

Reply:
@m4gr3d do you know if one way is better than the other?

Reply:
If everything is good to go with this PR, can it be approved? PR's #39997 and #40182 build on top of this.